parquet-column/src/main/java/org/apache/parquet/column/impl/ColumnWriteStoreV2.java

```diff
@@ -147,6 +148,11 @@ public void endRecord() {
   }
 
   private void sizeCheck() {
+    if(!parquetProperties.isEstimateNextSizeCheck()) {
```

Comment:
You're right.  I haven't really taken a look at v2 until now and missed that sizeCheck() is also doing the flush.

I'll revisit this.